### PR TITLE
Updated Main menu and resolving merge conflicts

### PR DESCRIPTION
**Pull request by jacobfrank4**

### CODE REVIEW

File: src/GameStateMenu.cpp [hunk 1]
```diff
@@ -10,143 +10,509 @@
...
 
     bool success = true;
 
-    //Open the font
-    frameFont = TTF_OpenFont( "assets/fonts/kenpixelsquare.ttf", 28 );
-    if ( frameFont == NULL ) {
-        printf( "Failed to load font! SDL_ttf Error: %s\n", TTF_GetError() );
-        success = false;
+    menuFont = TTF_OpenFont( "assets/fonts/Overdrive Sunset.otf", 110);
+    if ( menuFont == NULL ) {
```
Comment (isaacmorneau) [inline src/GameStateMenu.cpp]: `nullptr` should be used for checking nullness of pointers

File: src/GameStateMenu.cpp [hunk 2]
```diff
@@ -10,143 +10,509 @@
...
-        game.window.handleEvent(event);
-           switch( event.type ) {
-          case SDL_KEYDOWN:
-            switch( event.key.keysym.sym ) {
-            case SDLK_ESCAPE:
+	int x,y;
+	SDL_Keycode keyCode;
+
+	//Handle events on queue
+	SDL_WaitEvent( &event );
```
Comment (isaacmorneau) [inline src/GameStateMenu.cpp]: I actually had to look up the doc on this one, good choice!

File: src/GameStateMenu.cpp [hunk 3]
```diff
@@ -10,143 +10,509 @@
...
             break;
-        case SDL_QUIT:
-            play = false;
-            break;
-          default:
-            break;
-        }
-    }
+      	case SDL_KEYUP: //Do nothing on key release
+       		switch ( event.key.keysym.sym ) {
```
Comment (isaacmorneau) [inline src/GameStateMenu.cpp]: If keyup is ignored then there is no need for another switch inside
this should instead just be
```
case SDL_KEYUP: //Do nothing on key release
    break;
```

File: src/GameStateMenu.cpp [hunk 4]
```diff
@@ -10,143 +10,509 @@
...
+        		}
+        	}
+        	break;
+        case SDL_MOUSEBUTTONDOWN:
+        	x = event.button.x;
+        	y = event.button.y;
+        	for (size_t i = 0; i < NUM_MENU_ITEMS; i++) {
+                if (selected[i]) {
+                    update(i);
+        			play = false;
```
Comment (isaacmorneau) [inline src/GameStateMenu.cpp]: just noticed that this is tabs, please remove them.
for a quick solution there is a simple command [on the wiki](https://github.com/bcit-datacomm/4981_Linux_Game/wiki/Style-Guide#dont-use-tabs-ever-period-use-4-spaces) that you can run

File: src/GameStateMenu.cpp [hunk 5]
```diff
@@ -10,143 +10,509 @@
...
+                }
+        	}
+        	for (size_t i = 0; i < NUM_TEXT_FIELDS; i++) {
+                //Check if the mouse was clicked inside the textbox area
+        		if (x >= textboxPos[i].x && x <= textboxPos[i].x + textboxPos[i].w &&
+        			y >= textboxPos[i].y && y <= textboxPos[i].y + textboxPos[i].h) {
+                    activeTextbox[i] = true; //Activate the textbox to enable typing
+                    //If default string is detected, delete it
+                    if (!strcmp(textInput[i].c_str(), defaultText[i].c_str())) {
```
Comment (isaacmorneau) [inline src/GameStateMenu.cpp]: converting to c style strings then using strcmp works but doesnt utilize what was already made for strings see [string compare](http://www.cplusplus.com/reference/string/string/compare/)

File: src/GameStateMenu.cpp [hunk 6]
```diff
@@ -10,143 +10,509 @@
...
+                    if (textInput[i].empty()) {
+                        textInput[i] = defaultText[i];
+                    }
+        		}
+        	}
+
+        	break;
+        case SDL_WINDOWEVENT:
+        	switch (event.window.event) {
+	    		case SDL_WINDOWEVENT_RESIZED:
```
Comment (isaacmorneau) [inline src/GameStateMenu.cpp]: tabs and spaces on the same line? heh this is even worse

File: src/GameStateMenu.cpp [hunk 7]
```diff
@@ -10,143 +10,509 @@
...
+void GameStateMenu::positionElements() {
+
+    int         windowWidth     = game.window.getWidth();
+    int         windowHeight    = game.window.getHeight();
+    int         maxTextWidth    = 0;
+    int         maxTextHeight   = 0;
+    int         vertPadding     = 50;
+    int         horzPadding     = 10;
+    std::string longestString   = "";
+    char        largestChar     = 'W';
```
Comment (isaacmorneau) [inline src/GameStateMenu.cpp]: this isnt a blocking issue but for the future a single space between variable and name as well as name and value will be added to the style guide.

File: src/GameStateMenu.cpp [hunk 8]
```diff
@@ -10,143 +10,509 @@
...
+    }
+    //Calculate the pixel length and height of the largest possible string
+    longestString.resize(maxLength+1, largestChar);
+	TTF_SizeText(textboxFont, longestString.c_str(), &maxTextWidth, &maxTextHeight);
+
+    //Position the menu text
+    menuItemPos[JOIN].x     = windowWidth/2  - menuTextTextures[JOIN].getWidth()/2;
+    menuItemPos[JOIN].y     = windowHeight/2 + menuTextTextures[JOIN].getHeight();
+    menuItemPos[OPTIONS].x  = windowWidth/2  - menuTextTextures[OPTIONS].getWidth()/2;
+    menuItemPos[OPTIONS].y  = windowHeight/2 + menuTextTextures[OPTIONS].getHeight()*2.2;
```
Comment (isaacmorneau) [inline src/GameStateMenu.cpp]: in general try and space out operands and operators with a space between each ie 
```= windowWidth/2  - menuTextTextures[JOIN].getWidth()/2;```
is
```= windowWidth / 2  - menuTextTextures[JOIN].getWidth() / 2;```

File: src/GameStateMenu.cpp [hunk 9]
```diff
@@ -10,143 +10,509 @@
...
-
+	// Free texture and font
+	menuTextTextures[1].free();
+	menuTextTextures[0].free();
+	TTF_CloseFont(textboxFont);
+	TTF_CloseFont(headingFont);
+	TTF_CloseFont(menuFont);
+	menuFont = NULL;
+	textboxFont = NULL;
+	headingFont = NULL;
```
Comment (isaacmorneau) [inline src/GameStateMenu.cpp]: `nullptr` is better than `NULL` for pointers

**Review by isaacmorneau — CHANGES_REQUESTED** [review]
as per comments a few fixes are needed.

File: src/GameStateMenu.cpp (hunk 9 above)
Comment (jacobfrank4) [reply src/GameStateMenu.cpp]: resolved

File: src/GameStateMenu.cpp (hunk 8 above)
Comment (jacobfrank4) [reply src/GameStateMenu.cpp]: resolved

File: src/GameStateMenu.cpp (hunk 7 above)
Comment (jacobfrank4) [reply src/GameStateMenu.cpp]: resolved

File: src/GameStateMenu.cpp (hunk 6 above)
Comment (jacobfrank4) [reply src/GameStateMenu.cpp]: resolved

File: src/GameStateMenu.cpp (hunk 5 above)
Comment (jacobfrank4) [reply src/GameStateMenu.cpp]: resolved

File: src/GameStateMenu.cpp (hunk 4 above)
Comment (jacobfrank4) [reply src/GameStateMenu.cpp]: resolved

File: src/GameStateMenu.cpp (hunk 3 above)
Comment (jacobfrank4) [reply src/GameStateMenu.cpp]: resolved

File: src/GameStateMenu.cpp (hunk 2 above)
Comment (jacobfrank4) [reply src/GameStateMenu.cpp]: resolved

File: src/GameStateMenu.cpp (hunk 1 above)
Comment (jacobfrank4) [reply src/GameStateMenu.cpp]: resolved

**Review by isaacmorneau — APPROVED** [review]

**Pull request merged**